common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -4665,6 +4665,18 @@
           "Python UDF in the ON clause of a <joinType> JOIN. In case of an INNER JOIN consider rewriting to a CROSS JOIN with a WHERE clause."
         ]
       },
+      "QUERY_ONLY_CORRUPT_RECORD_COLUMN" : {
+        "message" : [
+          "Queries from raw JSON/CSV/XML files are disallowed when the",
```

Comment:
Here it is:
1. Because the current `XML` data source also has such behavior, I added XML here
https://github.com/apache/spark/blob/02795a36171cac51b1beaf68c42e8dfe3e6f0903/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/xml/XmlFileFormat.scala#L116-L118

2. Since the current changes in JSON, CSV, and XML are in different versions, I recommend removing `Since Spark 2.3` here.